src/gnn_reco/data/sqlite_dataconverter.py

```diff
@@ -233,15 +232,14 @@ def _processfiles(self):
             gcd_file_list = np.array_split(np.array(gcd_files),workers)
             for i in range(0,workers):
                 settings.append([file_list[i],str(i),gcd_file_list[i], event_nos[i], self.mode, self.pulsemap, self.max_dict_size, self.db_name, self.outdir])
-            if __name__ == 'sqlite_dataconverter':
-                #parallel_extraction(settings[0])
-                print('starting pool!')
-                p = Pool(processes = workers)
-                p.map_async(parallel_extraction, settings)
-                p.close()
-                p.join()
-                print('pool')
-                self._merge_databases()
+            
+            print('starting pool!')
+            p = Pool(processes = workers)
+            p.map_async(parallel_extraction, settings)
+            p.close()
+            p.join()
+            print('pool')
```

Comment:
As far as I can tell, `_process_files` is not called in any child process; therefore I don't know when this would be an issue. I have also run the script several times with no problems, so it seems like this should work. Otherwise I figure we would need to define a minimal working example (MWE) for recreating such a potential child-processes-spawning-child-processes problem?